### DISCUSSION TITLE
Developing bobheadxi/deployments@v1

### DISCUSSION THREAD

**Discussion by bobheadxi** [bobheadxi/deployments#36] [Ideas] [answered]
Surprised to see [so many usages of this action in the wild](https://sourcegraph.com/search?q=context:global+bobheadxi/deployments+file:%5E%5C.github/workflows+-repo:bobheadxi&patternType=literal)! Hoping to spend some time giving this action some love in the coming weeks.

Opening this up for discussion and notes on things that can be done for v1 🎏 

Milestone: https://github.com/bobheadxi/deployments/milestone/1

**Comment by bobheadxi** [comment]
Current outline:

* ability to use `uses: bobheadxi/deployments@v1`, which I'll probably do [via something like this](https://sourcegraph.com/github.com/bobheadxi/gobenchdata/-/blob/.github/workflows/publish_v1.yml#L16)
* refactor the steps and include some unit tests (#2), since it's starting to get rather convoluted
* improve the override paradigm: #25, #39
* #18 (testing)
* maybe have a prod at #19 as well

**Reply by bobheadxi** [reply]
Also https://github.com/bobheadxi/deployments/discussions/38

**Comment by bobheadxi** [comment]
FYI past contributors @shallwefootball , @naartjie , @V3lop5 , @AllexVeldman , @codyleblanc , @louy , @themightychris (thanks everyone!) and participants in issues (@zacklitzsinger, @Vadorequest, @johndesp, @gugaiz) in case any of you are still using this action and have any thoughts!

**Comment by V3lop5** [comment]
Hi there!

Maybe you could add a step to delete deployments completely. For that I'm currently using [strumwolf/delete-deployment-environment](https://github.com/strumwolf/delete-deployment-environment).

**Reply by bobheadxi** [reply]
This will be `step: delete-env` in https://github.com/bobheadxi/deployments/pull/40!

**Comment by bobheadxi** [comment, marked as answer]
`bobheadxi/deployments@v1` is now available! https://github.com/bobheadxi/deployments/discussions/82
